### ISSUE TITLE
Issues with grpc-web 1.0.7 + envoy 1.13 + Safari 13

### ISSUE THREAD

**Issue by phlippieb** [grpc/grpc-web#759]
Hi!

My team uses grpc-web at the latest version (1.0.7). We recently upgraded our server to use Envoy 1.13, and with this upgrade, Safari could no longer receive functional responses from Envoy. Other browsers were unaffected. We then downgraded Envoy to 1.12, and the issue was resolved. So there seems to be an incompatibility with:
- grpc-web 1.0.7
- envoy 1.13
- safari 13 (did not test other versions)

We also noticed that last release of grpc-web was made in November 2019, and there have been some commits made since then that _might_ address this compatibility issue. I tried to point our web app to grpc-web's master branch, but npm makes this rather difficult if the package.json is not in the root :) so I gave up after a while.

So just wanted to check:
- Is this something you're aware of?
- Do you plan to release a new version to npm soon?

**Comment by stanley-cheung** [comment]
We are not aware of Safari-specific issue. But this sounded like an Envoy issue. In the meantime we do plan to cut a next release soon as there are indeed quite a few changes we merged recently.

**Comment by yoitsro** [comment]
Hi @phlippieb. I'm seeing this behaviour as well. The request succeeds, but in the inspector in Safari, this is the error I see:

`Failed to load resource: WebKit encountered an internal error`

#### Summary
```
URL: https://myurl.io/svc.Authentication/Login
Status: —
Source: —
Initiator: 
2.c85f5bc6.chunk.js:2:583533
```

#### Request
```
Content-Type: application/grpc-web-text
Accept: application/grpc-web-text
Origin: https://develop--elastic-jennings-2e862a.netlify.com
Referer: https://develop--elastic-jennings-2e862a.netlify.com/login
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.5 Safari/605.1.15
X-User-Agent: grpc-web-javascript/0.1
X-Grpc-Web: 1
grpc-timeout: 4999m
```

#### Response
```
No response headers
```

#### Request Data
```
MIME Type: application/grpc-web-text
Request Data: 
```

**Comment by phlippieb** [comment]
@yoitsro Yes, that's my exact experience. Downgrading Envoy to 1.12 fixed it for us.

**Comment by stanley-cheung** [comment]
Perhaps it makes sense to log a bug to Envoy to alert them of this regression?

**Comment by yoitsro** [comment]
I'll do that now.

**Comment by phlippieb** [comment]
I see @yoitsro opened an issue on Envoy's repo. Thanks! https://github.com/envoyproxy/envoy/issues/10514

**Comment by ussuritiger** [comment]
Hey @yoitsro, any update on this?

**Comment by yoitsro** [comment]
Hey @ussuritiger. The latest is that this is a bug with WebKit which has been filed here https://bugs.webkit.org/show_bug.cgi?id=210108, which has resulted in a bug being tracked on Apple's internal tracker via their Feedback Assistant: FB7654794

**Comment by ussuritiger** [comment]
Clear, thanks @yoitsro. Still wondering what has been changed on Envoy side that caused WebKit to have a problem.

**Comment by yoitsro** [comment]
Yeah, it'd be interesting to compare the actual packets. I'd love to do this, but time is limited right now :(

**Comment by engineerdev** [comment]
Set the parameters:
alpn_protocols:
- "http/1.1"
- "h2"

**Comment by rogchap** [comment]
We are experiencing the same issues, but also note that it seems to effect certain versions of Windows 10 and Internet Explorer/Edge (pre Chromium versions) as well as Safari.

@engineerdev 's fix to enable ALPN for HTTP/1.1 does indeed work; but requests are now forced to use HTTP/1.1 even when HTTP/2 is available; so this workaround is not a long term solution.

You may want to downgrade to Envoy 1.12 until a full (HTTP/2) solutions is available.

**Comment by jwvictor** [comment]
@rogchap  I just ran into this as well. I was able to make it go away by proxying through nginx, but like @engineerdev this forced me into http/1.1. I'm using Gloo so Envoy was presumably installed in the background. Any ideas what I should do here?

**Comment by engineerdev** [comment]
envoy 1.15.0 the same problem :(

**Comment by yulin-liang** [comment]
Looks like **Envoy** has had a PR to fix it. [#12178](https://github.com/envoyproxy/envoy/pull/12178)

**Comment by yoitsro** [comment]
This PR has now been merged 🎉 https://github.com/envoyproxy/envoy/pull/12178

I guess it'll be going into the next release.
